src/bgw/job_stat_history.c

```diff
@@ -21,51 +23,107 @@ typedef struct BgwJobStatHistoryContext
 	Jsonb *edata;
 } BgwJobStatHistoryContext;
 
+static Jsonb *
+build_job_info(BgwJob *job)
+{
+	JsonbParseState *parse_state = NULL;
+	pushJsonbValue(&parse_state, WJB_BEGIN_OBJECT, NULL);
+
+	/* all fields that is possible to change with `alter_job` API */
+	ts_jsonb_add_interval(parse_state, "schedule_interval", &job->fd.schedule_interval);
+	ts_jsonb_add_interval(parse_state, "max_runtime", &job->fd.max_runtime);
+	ts_jsonb_add_int32(parse_state, "max_retries", job->fd.max_retries);
+	ts_jsonb_add_interval(parse_state, "retry_period", &job->fd.retry_period);
+	ts_jsonb_add_str(parse_state, "proc_schema", NameStr(job->fd.proc_schema));
+	ts_jsonb_add_str(parse_state, "proc_name", NameStr(job->fd.proc_name));
+	ts_jsonb_add_str(parse_state, "owner", GetUserNameFromId(job->fd.owner, false));
+	ts_jsonb_add_bool(parse_state, "scheduled", job->fd.scheduled);
+	ts_jsonb_add_bool(parse_state, "fixed_schedule", job->fd.fixed_schedule);
+
+	if (job->fd.initial_start)
+		ts_jsonb_add_interval(parse_state, "initial_start", &job->fd.retry_period);
+
+	if (job->fd.hypertable_id != INVALID_HYPERTABLE_ID)
+		ts_jsonb_add_int32(parse_state, "hypertable_id", job->fd.hypertable_id);
+
+	if (job->fd.config != NULL)
+	{
+		/* config information jsonb*/
+		JsonbValue value = { 0 };
+		JsonbToJsonbValue(job->fd.config, &value);
+		ts_jsonb_add_value(parse_state, "config", &value);
+	}
+
+	if (strlen(NameStr(job->fd.check_schema)) > 0)
+		ts_jsonb_add_str(parse_state, "check_schema", NameStr(job->fd.check_schema));
+
+	if (strlen(NameStr(job->fd.check_name)) > 0)
+		ts_jsonb_add_str(parse_state, "check_name", NameStr(job->fd.check_name));
+
+	if (job->fd.timezone != NULL)
+		ts_jsonb_add_str(parse_state, "timezone", text_to_cstring(job->fd.timezone));
+
+	return JsonbValueToJsonb(pushJsonbValue(&parse_state, WJB_END_OBJECT, NULL));
+}
+
+static Jsonb *
+ts_bgw_job_stat_history_build_data_info(BgwJobStatHistoryContext *context)
+{
+	JsonbParseState *parse_state = NULL;
+	JsonbValue value = { 0 };
+	pushJsonbValue(&parse_state, WJB_BEGIN_OBJECT, NULL);
+
+	Assert(context->job != NULL);
```

Comment:
Nit: It is not needed to prevent execution (code after will not work with `context` being NULL anyway), but it communicates intentions clearly.

```suggestion
	Assert(context != NULL && context->job != NULL);
```

Reply:
I also like to better communicate intentions. Done!